src/mumble/CMakeLists.txt

```
@@ -416,9 +416,13 @@ else()
 	endif()
 endif()
 
+# This is used to invalidate the saved widget state if a change in the source is detected.
+file(SHA3_256 "${CMAKE_CURRENT_SOURCE_DIR}/MainWindow.ui" MAINWINDOW_UI_HASH)
+
 target_compile_definitions(mumble_client_object_lib
 	PUBLIC
 		"MUMBLE"
+		"MAINWINDOW_UI_HASH=\"${MAINWINDOW_UI_HASH}\""
```

```suggestion
		"MUMBLE_MAINWINDOW_UI_HASH=\"${MAINWINDOW_UI_HASH}\""
```
why are you passing the hash as a string instead of a number?

---

Hashes are commonly represented in hexadecimal format, but they're not numerical.

For example, let's look at the current SHA3-256 hash of `MainWindow.ui`:

```
71dff43a23dd3b08af155f3272373f729abf52532c69a012dbe47170fdaef367
```

Every 2 characters represent a byte, for a total of 32 with this algorithm (256 bits).

---

Right. You can convert it to a number directly in cmake though: https://cmake.org/cmake/help/latest/command/math.html (input as hex and set the output format to decimal)

---

Good catch. Also splitting the string (as we need to convert byte-per-byte) seems to be straightforward using a regular expression.

---

We don't have to convert byte by byte. The hex conversion should do all of that for us, no?

---

You would end up with an insanely high decimal number, if the conversion actually succeeds. We could truncate/round the number to fit a 32 bit integer on the C++ side in that case, but it probably wouldn't make much sense.